src/interface/variable.hpp

```diff
@@ -282,7 +282,15 @@ class ParticleVariable {
   }
 
   KOKKOS_FORCEINLINE_FUNCTION
-  auto GetDim(const int i) const { return data.GetDim(i); }
+  auto GetDim(const int i) const {
+    PARTHENON_REQUIRE(0 < i && i <= 6, "ParArrayNDGenerics are max 6D");
```

Comment:
How about `PARTHENON_DEBUG_REQUIRE` (similar to the assert in for the ParArrayND itself)?

Reply:
Agreed, fixed.